drivers/clickhouse/reader.go

```diff
@@ -0,0 +1,213 @@
+package clickhouse
+
+import (
+	"database/sql"
+	"strings"
+
+	"github.com/xo/usql/drivers"
+	"github.com/xo/usql/drivers/metadata"
+)
+
+type MetadataReader struct {
+	metadata.LoggingReader
+}
+
+// NewMetadataReader creates the metadata reader for clickhouse databases.
+func NewMetadataReader(db drivers.DB, opts ...metadata.ReaderOption) metadata.Reader {
+	return &MetadataReader{
+		LoggingReader: metadata.NewLoggingReader(db, opts...),
+	}
+}
+
+var (
+	_ metadata.BasicReader    = &MetadataReader{}
+	_ metadata.FunctionReader = &MetadataReader{}
+)
+
+func (r MetadataReader) Tables(f metadata.Filter) (*metadata.TableSet, error) {
+	qstr := `SELECT 
+	database AS Schema,
+	name AS Name,
+	COALESCE(
+		IF(database LIKE 'system', 'SYSTEM TABLE', null),
+		IF(is_temporary,'LOCAL TEMPORARY', null),
+		IF(engine LIKE 'View', 'VIEW', null), 
+		'TABLE'
+	) AS Type,
+	COALESCE(total_bytes, 0) AS Size,
+	comment as Comment 
+FROM 
+	system.tables`
+	conds := []string{}
+	vals := []interface{}{}
+	if !f.WithSystem {
+		conds = append(conds, "database NOT LIKE 'system'")
+	}
+	if f.Schema != "" {
+		vals = append(vals, f.Schema)
+		conds = append(conds, "database LIKE ?")
+	}
+	if f.Name != "" {
+		vals = append(vals, f.Name)
+		conds = append(conds, "name LIKE ?")
+	}
+	if len(f.Types) != 0 {
+		pholders := []string{}
+		for _, t := range f.Types {
+			vals = append(vals, t)
+			pholders = append(pholders, "?")
+		}
+		if len(pholders) != 0 {
+			conds = append(conds, "Type IN ("+strings.Join(pholders, ", ")+")")
+		}
+	}
+	rows, closeRows, err := r.query(qstr, conds, "Schema, Name", vals...)
+	if err != nil {
+		return nil, err
+	}
+	defer closeRows()
+
+	results := []metadata.Table{}
+	for rows.Next() {
+		rec := metadata.Table{}
+		err = rows.Scan(&rec.Schema, &rec.Name, &rec.Type, &rec.Size, &rec.Comment)
+		if err != nil {
+			return nil, err
+		}
+		results = append(results, rec)
+	}
+	if rows.Err() != nil {
+		return nil, rows.Err()
+	}
+	return metadata.NewTableSet(results), nil
+}
+
+func (r MetadataReader) Columns(f metadata.Filter) (*metadata.ColumnSet, error) {
+	qstr := `SELECT
+  		position,
+  		database as schema,
+  		name,
+  		type,
+  		COALESCE(default_expression, '')
+	FROM 
```

Comment:
Nit: queries should be aligned to the start of line, not to the surrounding code.